File: src/0832-Flipping-an-Image/0832.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& A) {
        int n = A.size();
        for (int i = 0; i < n; i++) {
            int l = 0, r = A[0].size() - 1;
            while (l < r) {
                if (A[i][l] == A[i][r]) {
                    A[i][l] ^= 1;
                    A[i][r] ^= 1;
                } 
                l++, r--;
            }
            if (l == r) {
                A[i][l] ^= 1;
            }
        }
        return A;
    }
};

File: src/0832-Flipping-an-Image/0832.py
class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        arr = list()
        for row in image:
            arr.append(row[::-1])
            for i in range(len(arr[-1])):
                if arr[-1][i] == 0:
                    arr[-1][i] = 1
                else:
                    arr[-1][i] = 0
        return arr
